keps/sig-storage/3333-reconcile-default-storage-class/README.md

```diff
@@ -0,0 +1,656 @@
+
+Risk: Users depend on existing Kubernetes behavior.
+
+Mitigation:
+
+* Document the behavior change with a release note.
+
+* Suggest users that expect to bind to PVs with `storageClassName=""` to create
+  PVCs with `storageClassName=""` and not `storageClassName=nil` in Kubernetes
+  docs.
+
+## Design Details
+
+This KEP requires changes in:
+* kube-controller-manager / its PV controller:
+    * Binding of PVCs with `pvc.spec.storageClassName=nil` must be changed to
+      ignore PVs with `pv.spec.storageClassName=""`.
+    * `pvc.spec.storageClassName=nil` in PVC must be reconciled to a current
+      default SC, if it exists, or after it's created.
+
+* kube-apiserver / PVC update validations:
+    * PVC update from  `pvc.spec.storageClassName=nil` to
+      `pvc.spec.storageClassName=<storage class name>` is now forbidden, and it
+      must be allowed for this KEP to work.
+
+### Test Plan
+
+[x] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+Kubernetes already has a good coverage of PV/PVC binding tests in
+[test/e2e/storage/persistent_volumes.go](https://github.com/kubernetes/kubernetes/blob/ccfac6d3200f63656575819e7b5976c12c3019a6/test/e2e/storage/persistent_volumes.go)
+
+There are no e2e tests that cover behavior of the default SC presence / absence,
+they will be added only with the new behavior.
+
+##### Unit tests
+
+All changes should be only in the packages below, which have enough unit test
+coverage, we will add only unit tests for the new or changed code.
+
+- `pkg/controller/volume/persistentvolume`: 2022-06-03 - 79%
+- `pkg/apis/core/validation/`: 2022-06-03 - 82%
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- test/integration/volume/persistent_volumes_test.go: Has few cases for PV/PVC
+  binding.
+
+We plan to extend these test to include default SC and how it will be applied to
+existing PVCs. We use integration tests almost like e2e tests of the new
+behavior to be sure that change of the default SC won't affect other tests
+running in parallel.
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+There are no tests that would check how the default SC, because it would need to
```

Comment:
how the default SC -> how the default SC works

Reply:
Fixed.